torch/_inductor/ir.py

```diff
@@ -6460,6 +6460,36 @@ def codegen(self, wrapper):
         )
 
 
+class Broadcast(InPlaceCollectiveKernel):
```

Comment:
@yifuwang @Chillee wanted to clarify the behavior of InplaceCollective

broadcast is a little different from e.g. allreduce_, in the sense that (a) it is technically 'inplace' bc it reads or writes to an input tensor depending on rank, but (b) the tensor it writes to is assumed to be empty, unlike allreduce where it's a read-modify-write.